client/modules/cm-lang-ftml/src/tips/BlockTip.svelte

```diff
@@ -0,0 +1,242 @@
+<!--
+  @component Autocomplete tooltip for FTML blocks.
+-->
+<script lang="ts">
+  import type { Block } from "../data/types"
+  import type { EditorSvelteComponentProps } from "sheaf-core"
+  import * as Prism from "wj-prism"
+  import { t } from "wj-state"
+  import type { FTMLFragment } from "ftml-wasm-worker"
+  import { Icon, TippySingleton } from "components"
+
+  interface Docs {
+    title: string
+    info: FTMLFragment
+    example: string
+  }
+
+  export let name: string
+  export let block: Block
+  export let docs: Docs
+  export let unmount: EditorSvelteComponentProps["unmount"]
+
+  // TODO: deprecated styling (need a deprecated block first?)
+
+  const aliases = [name, ...(block.aliases ?? [])]
+  const deprecated = block["deprecated"] ?? false
+  const acceptsStar = block["accepts-star"] ?? false
+  const acceptsScore = block["accepts-score"] ?? false
+  const acceptsNewlines = block["accepts-newlines"] ?? false
+  const usesHTMLAttributes = block["html-attributes"] ?? false
+  const [outputType, outputTag, outputClass] = block["html-output"].split(",")
+
+  let codeString =
+    outputType === "html"
+      ? outputClass
+        ? `<${outputTag} class="${outputClass}">`
+        : `<${outputTag}>`
+      : `type: ${outputType}`
+
+  codeString = Prism.highlight(codeString, outputType === "html" ? "html" : "log")
+
+  // looks messy but this just assembles a reasonable looking block string
+  let ftmlString = `[[${name}${
+    block.head === "map"
+      ? ' arg="value"'
+      : block.head === "value"
+      ? " value"
+      : block.head === "value+map"
+      ? ' value arg="value"'
+      : ""
+  }]]`
```

Comment:
Maybe? Switch isn't an expression, so I'd have to have a temp variable or assemble the string the same way in every case, or something like that. I tend to avoid switch because its syntax sucks.